rosie/src/main/java/com/karumi/rosie/domain/usecase/UseCaseHandler.java

```diff
@@ -29,6 +33,10 @@
   private final TaskScheduler taskScheduler;
   private final ErrorHandler errorHandler;
 
+  // This list is used to retain error adapters in a non-weak reference associated with this use
+  // case handler
+  private final List<OnErrorCallback> errorCallbacks = new ArrayList<>();
```

Comment:
I had the same idea, about this can leak the activity, but I'm checking and usecasaehandler is an weak reference too, no? can you do a double ckeck for me?